crates/wasmparser/src/readers/core/types.rs

```diff
@@ -83,13 +83,13 @@ fn can_fit_max_wasm_types_in_packed_index() {
 
 impl PackedIndex {
     const UNUSED_MASK: u32 = u32::MAX & !(Self::KIND_MASK | Self::INDEX_MASK);
-    const KIND_MASK: u32 = 0b11 << 20;
-    const INDEX_MASK: u32 = (1 << 20) - 1;
+    const KIND_MASK: u32 = 0b11 << 19;
+    const INDEX_MASK: u32 = (1 << 19) - 1;
 
-    const MODULE_KIND: u32 = 0b00 << 20;
-    const REC_GROUP_KIND: u32 = 0b01 << 20;
+    const MODULE_KIND: u32 = 0b00 << 19;
+    const REC_GROUP_KIND: u32 = 0b01 << 19;
     #[cfg(feature = "validate")]
-    const ID_KIND: u32 = 0b10 << 20;
+    const ID_KIND: u32 = 0b10 << 19;
```

Comment:
For these changes mind updating the documentation on `PackedIndex` above? Also I think that `0b11` as a "kind" is open to represent "exact" so the index bit-space need not be shrunk?